docs/data/material/guides/next-js-app-router/next-js-app-router.md

```diff
@@ -0,0 +1,227 @@
+
+By default emotion will inject Material UI styles at the bottom of the HTML `<head>`, which gives it precedence over custom styles (e.g. from CSS modules, Tailwind CSS, or even plain CSS).
+
+In regular client-side React, the `prepend: true` option is passed to `createCache` to reverse the injection order, so custom styles can override Material UI styles without using `!important`.
+
+Currently `prepend` does not work reliably with the App Router, but you can work around it by wrapping emotion styles in a CSS `@layer` with a modification to the snippet above:
+
+```diff
+ useServerInsertedHTML(() => {
+   const names = flush();
+   if (names.length === 0) {
+     return null;
+   }
+   let styles = '';
+   for (const name of names) {
+     styles += cache.inserted[name];
+   }
+   return (
+     <style
+       key={cache.key}
+       data-emotion={`${cache.key} ${names.join(' ')}`}
+       dangerouslySetInnerHTML={{
+-        __html: styles,
++        __html: options.prepend ? `@layer emotion {${styles}}` : styles,
+       }}
+     />
+   );
+ });
+```
+
+## Using Joy UI
+
+The set up for Joy UI is the same as [Material UI with a custom theme](#using-material-ui-with-a-custom-theme), except replace Material's `ThemeProvider` with Joy UI's `CssVarsProvider`:
+
+```diff
+ // ThemeRegistry
+-import { ThemeProvider } from '@mui/material/styles';
++import { CssVarsProvider } from '@mui/joy/styles';
+```
+
+See this [example repo](https://github.com/mui/material-ui/blob/master/examples/joy-next-app-router-ts) for a working demo of Joy UI with the Next.js App Router.
+
+## Using Base UI
+
+Base UI is styling-solution-agnostic, if you are using Emotion you can follow the same steps above.
+
+If you are using Tailwind CSS, see this [example repo](https://github.com/mui/material-ui/blob/master/examples/base-next-app-router-tailwind-ts) for a working demo.
+
+### Customizing Base UI with function props
+
+A common customization method in Base UI is to pass a function to slots in `slotProps` in order to apply dynamic props to slots. For example, changing the background color by applying a different class when a Button is disabled:
+
+```tsx
+// page.tsx
+
+export default function Page() {
+  return (
+    <>
+      {/* Next.js won't render this button without 'use-client'*/}
+      <Button
+        slotProps={{
+          root: (ownerState: ButtonOwnerState) => ({
+            className: ownerState.disabled ? 'bg-gray-400' : 'bg-blue-400',
+          }),
+        }}
+      >
+        Submit
+      </Button>
+
+      {/* Next.js can render this */}
+      <Button
+        slotProps={{
+          root: {
+            className: 'bg-gray-400',
+          },
+        }}
+      >
+        Return
+      </Button>
+    </>
+  );
+}
+```
+
+However, this will not work since function props are [non-serializable](https://nextjs.org/docs/getting-started/react-essentials#passing-props-from-server-to-client-components-serialization).
```

Comment:
```suggestion
Unfortunately, **this will not work** since function props are [non-serializable](https://nextjs.org/docs/getting-started/react-essentials#passing-props-from-server-to-client-components-serialization).
```